activemq-client-jakarta/src/main/java/org/apache/activemq/command/ActiveMQMessage.java

```diff
@@ -0,0 +1,808 @@
+        setObjectProperty(name, Boolean.valueOf(value), checkReadOnly);
+    }
+
+    @Override
+    public void setByteProperty(String name, byte value) throws JMSException {
+        setObjectProperty(name, Byte.valueOf(value));
+    }
+
+    @Override
+    public void setShortProperty(String name, short value) throws JMSException {
+        setObjectProperty(name, Short.valueOf(value));
+    }
+
+    @Override
+    public void setIntProperty(String name, int value) throws JMSException {
+        setObjectProperty(name, Integer.valueOf(value));
+    }
+
+    @Override
+    public void setLongProperty(String name, long value) throws JMSException {
+        setObjectProperty(name, Long.valueOf(value));
+    }
+
+    @Override
+    public void setFloatProperty(String name, float value) throws JMSException {
+        setObjectProperty(name, Float.valueOf(value));
+    }
+
+    @Override
+    public void setDoubleProperty(String name, double value) throws JMSException {
+        setObjectProperty(name, Double.valueOf(value));
+    }
+
+    @Override
+    public void setStringProperty(String name, String value) throws JMSException {
+        setObjectProperty(name, value);
+    }
+
+    protected void checkReadOnlyProperties() throws MessageNotWriteableException {
+        if (readOnlyProperties) {
+            throw new MessageNotWriteableException("Message properties are read-only");
+        }
+    }
+
+    protected void checkReadOnlyBody() throws MessageNotWriteableException {
+        if (readOnlyBody) {
+            throw new MessageNotWriteableException("Message body is read-only");
+        }
+    }
+
+    public Callback getAcknowledgeCallback() {
+        return acknowledgeCallback;
+    }
+
+    public void setAcknowledgeCallback(Callback acknowledgeCallback) {
+        this.acknowledgeCallback = acknowledgeCallback;
+    }
+
+    /**
+     * Send operation event listener. Used to get the message ready to be sent.
+     */
+    public void onSend() throws JMSException {
+        setReadOnlyBody(true);
+        setReadOnlyProperties(true);
+    }
+
+    @Override
+    public Response visit(CommandVisitor visitor) throws Exception {
+        return visitor.processMessage(this);
+    }
+
+    @Override
+    public void storeContent() {
+    }
+
+    @Override
+    public void storeContentAndClear() {
+        storeContent();
+    }
+
+    @Override
+    protected boolean isContentMarshalled() {
+        //Always return true because ActiveMQMessage only has a content field
+        //which is already marshalled
+        return true;
+    }
+
+    @Override
+    public long getJMSDeliveryTime() throws JMSException {
+        throw new UnsupportedOperationException("getJMSDeliveryTime() is unsupported in transitional client");
+    }
+
+    @Override
+    public void setJMSDeliveryTime(long deliveryTime) throws JMSException {
+        throw new UnsupportedOperationException("setJMSDeliveryTime(long) is unsupported in transitional client");
```

Comment:
This has been implemented in Main, this PR should be updated to include the latest changes from Main since this will be targeted for 5.18.0 now